### DISCUSSION TITLE
Design: Columns are in session instead of URL like the rest

### DISCUSSION THREAD

**Discussion by inmanturbo** [rappasoft/laravel-livewire-tables#747] [Q&A]
@rappasoft I'm just curious what the reasoning is behind the design choice to put the selected columns into the session instead of the query string. Would it not be nice for users to be able to share a link to a table with desired columns selected? And would it be any more difficult to put them into the query string like the searches and filters?

**Comment by rappasoft** [comment]
That's a good question and I don't have a good answer other than I assumed the user would want them saved and not have to reselect the columns they want every time they come back. There's no reason we cant have both though, populate the query string from the session or something, but then what does it do If you send it to someone that has a different configuration?

**Reply by inmanturbo** [reply]
Hmm. Good question as well. I think a good way might be to check the request first and use that to set/override the session if there is a valid "fields"  key in the request. That way when the come back to a "fresh" table (ie with no query string) they will have whatever columns they last had selected, yet they can still "bookmark" configurations and come back to them in whatever state the had set at the time they copied the URL.

**Reply by inmanturbo** [reply]
Granted "fields" may not be the right key, as it is often reserved for the actual builder query. Example: 

https://spatie.be/docs/laravel-query-builder/v5/features/selecting-fields#content-basic-usage

Something like "columns[table]" might be better.

**Reply by rappasoft** [reply]
Yeah i'm not fully thrilled with the way the query stuff came out again for V2,  but I just don't have the knowledge to do it any better right now I don't think. I like the overrides idea though.

**Reply by inmanturbo** [reply]
I'm not being critical of the query stuff, it's serviceable and a lot of ppl are happy with it. lol. Besides it can always be iterated on as you find ways that you like better. I was just thinking about an idea for being able to `deep link` the visible column selection, and I'm grateful you found the time to discuss it a bit. :smile: 

Really what I meant was, you know your api better than I do, and would probably know better than I would what key should be used in the query string for the "visible columns" list, also to avoid potential collisions in the future if you have any plans for making changes to the query request api.

**Reply by rappasoft** [reply]
Yeah definitely, I'm going to think about it and find the best way to support it without breaking anything lol
